apis/kueue/webhooks/resourceflavor_webhook.go

```diff
@@ -66,23 +70,92 @@ var _ webhook.CustomValidator = &ResourceFlavorWebhook{}
 func (w *ResourceFlavorWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
 	rf := obj.(*kueue.ResourceFlavor)
 	resourceFlavorLog.V(5).Info("Validating create", "resourceFlavor", klog.KObj(rf))
-	return ValidateResourceFlavorLabels(rf).ToAggregate()
+	return ValidateResourceFlavor(rf).ToAggregate()
 }
 
 // ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
 func (w *ResourceFlavorWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
 	newRF := newObj.(*kueue.ResourceFlavor)
 	resourceFlavorLog.V(5).Info("Validating update", "resourceFlavor", klog.KObj(newRF))
-	return ValidateResourceFlavorLabels(newRF).ToAggregate()
+	return ValidateResourceFlavor(newRF).ToAggregate()
 }
 
 // ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
 func (w *ResourceFlavorWebhook) ValidateDelete(ctx context.Context, obj runtime.Object) error {
 	return nil
 }
 
-func ValidateResourceFlavorLabels(rf *kueue.ResourceFlavor) field.ErrorList {
-	field := field.NewPath("labels")
-	return validation.ValidateLabels(rf.Labels, field)
+func ValidateResourceFlavor(rf *kueue.ResourceFlavor) field.ErrorList {
+	var allErrs field.ErrorList
 
+	labelsPath := field.NewPath("labels")
+	if len(rf.Labels) > 8 {
+		allErrs = append(allErrs, field.Invalid(labelsPath, rf.Labels, "must have at most 8 elements"))
+	}
+	allErrs = append(allErrs, metavalidation.ValidateLabels(rf.Labels, labelsPath)...)
+
+	taintsPath := field.NewPath("taints")
+	if len(rf.Taints) > 8 {
+		allErrs = append(allErrs, field.Invalid(taintsPath, rf.Taints, "must have at most 8 elements"))
+	}
+	allErrs = append(allErrs, validateNodeTaints(rf.Taints, taintsPath)...)
+	return allErrs
+}
+
+// validateNodeTaints is extracted from git.k8s.io/kubernetes/pkg/apis/core/validation/validation.go
+func validateNodeTaints(taints []corev1.Taint, fldPath *field.Path) field.ErrorList {
```

Comment:
Place this and `validateTaintEffect` under common.go makes more sense to me since it's extracted from kubernetes.